File: LevelEditor/code/models/GridRect.ts
import { Point } from "./point";
import { Shape } from "./Shape";
import { GridCoords } from "./GridCoords";
import { Rect } from "./Rect";
import { consts } from "../consts";

export class GridRect {
  
  topLeftGridCoords: GridCoords;
  botRightGridCoords: GridCoords;

  constructor(i1: number, j1: number, i2: number, j2: number) {
    this.topLeftGridCoords = new GridCoords(i1, j1);
    this.botRightGridCoords = new GridCoords(i2, j2);
  }
  toString() {
    return String(this.topLeftGridCoords.i) + "_" + String(this.topLeftGridCoords.j) + "_" + String(this.botRightGridCoords.i) + "_" + String(this.botRightGridCoords.j);
  }
  equals(other: GridRect) {
    return this.topLeftGridCoords.equals(other.topLeftGridCoords) && this.botRightGridCoords.equals(other.botRightGridCoords);
  }
  getRect() {
    return new Rect(this.j1 * consts.TILE_WIDTH, this.i1 * consts.TILE_WIDTH, (this.j2 + 1) * consts.TILE_WIDTH, (this.i2 + 1) * consts.TILE_WIDTH);
  }
  get i1() { return this.topLeftGridCoords.i; }
  get j1() { return this.topLeftGridCoords.j; }
  get i2() { return this.botRightGridCoords.i; }
  get j2() { return this.botRightGridCoords.j; }
}

File: GameEditor/GameEditor/Models/GridRect.cs
﻿using GameEditor.Editor;

namespace GameEditor.Models
{
    public class GridRect
    {
        public GridCoords topLeftGridCoords { get; set; }
        public GridCoords botRightGridCoords { get; set; }

        public GridRect(int i1, int j1, int i2, int j2)
        {
            this.topLeftGridCoords = new GridCoords(i1, j1);
            this.botRightGridCoords = new GridCoords(i2, j2);
        }

        public string toString()
        {
            return (this.topLeftGridCoords.i).ToString() + "_" + (this.topLeftGridCoords.j).ToString() + "_" + (this.botRightGridCoords.i).ToString() + "_" + (this.botRightGridCoords.j).ToString();
        }

        public bool equals(GridRect other)
        {
            return this.topLeftGridCoords.equals(other.topLeftGridCoords) && this.botRightGridCoords.equals(other.botRightGridCoords);
        }

        public Rect getRect()
        {
            return new Rect(this.j1 * Consts.TILE_WIDTH, this.i1 * Consts.TILE_WIDTH, (this.j2 + 1) * Consts.TILE_WIDTH, (this.i2 + 1) * Consts.TILE_WIDTH);
        }

        public int i1 { get { return this.topLeftGridCoords.i; } }
        public int j1 { get { return this.topLeftGridCoords.j; } }
        public int i2 { get { return this.botRightGridCoords.i; } }
        public int j2 { get { return this.botRightGridCoords.j; } }
    }
}
